common/src/event_logging.rs

```diff
@@ -0,0 +1,47 @@
+mod utils;
+pub use utils::*;
+
+#[xmtp_macro::build_logging_metadata]
+pub enum Event {
```

Comment:
This is the new thing. You can now:

```rust
log_event!(
    Event::CreatedGroup,
    group_id = ?hex::encode(&group_id),
    other_structured_data_that_will_be_included_in_the_log = 123,
);
```

And it will log that event with that important context data whether you're in structured or human-readable logs (while de-duping). It also requires that those context fields be present in the log, providing a nice error message when they're missing.